src/charm.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+import logging
+
+from ops.main import main
+from ops.pebble import Layer
+from ops.charm import CharmBase
+from ops.model import ActiveStatus, WaitingStatus, MaintenanceStatus
+
+logger = logging.getLogger(__name__)
+
+
+class TrainingOperatorCharm(CharmBase):
+    """ A Juju Charm for Training Operator"""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        self.logger = logging.getLogger(__name__)
+
+        self._name = self.model.app.name
+        self._namespace = self.model.name
+        self._manager_service = "manager"
+
+        self.framework.observe(self.on.install, self._on_config_changed)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.training_operator_pebble_ready, self._on_training_operator_pebble_ready)
+
+    def _training_operator_layer(self):
+        """
+        Returns a Pebble configuration layer for Kubeflow Training Operator
+        This piece of software takes care of the deployment part of the
+        training-operator
+        """
```

Comment:
We generally use the Google docstring format for operator framework and charms. This docstring might look like the following using that format:

```suggestion
        """Property that returns a pre-configured Pebble layer.
        
        The Pebble layer contains information about how to start the manager service,
        and sets the relevant environment variables for the process to start successfully.
        
        Returns:
            dict: a dictionary representation of a Pebble layer
        """
```

Note:
    - First line is on the same line as the opening `"""`, ends with a full stop
    - The second section allows you to add more context if needed (though probably isn't here)
    - There should be seperate sections for `Args`, `Raises`, `Returns`.

There are a bunch more examples here: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
+1, will use it from now on.